### PR TITLE
small tweaks and one bugfix to location/user behavior

### PR DESCRIPTION
**Pull request by czue**
http://manage.dimagi.com/default.asp?165439 is fixed in https://github.com/dimagi/commcare-hq/commit/a13ba09b4c48707346f36f71ec4306002d01111a

the rest is optimizations and cleanup.

@kaapstorm cc @esoergel

### CODE REVIEW

File: corehq/apps/users/models.py [hunk 1]
```diff
@@ -1320,241 +1320,7 @@ def fn(domain=None, data=None):
             self.__class__.__name__, item))
 
 
-class LocationUserMixin(DocumentSchema):
```
Comment (czue) [inline corehq/apps/users/models.py]: this mixin was used in exactly one place which seemed silly so i just merged it with the user moel

File: corehq/apps/users/views/mobile/users.py [hunk 2]
```diff
@@ -79,11 +78,12 @@ def dispatch(self, request, *args, **kwargs):
     @property
     @memoized
     def custom_data(self):
+        is_custom_data_post = self.request.method == "POST" and self.request.POST['form_type'] == "update-user"
         return CustomDataEditor(
             field_view=UserFieldsView,
             domain=self.domain,
             existing_custom_data=self.editable_user.user_data,
-            post_dict=self.request.POST if self.request.method == "POST" else None,
+            post_dict=self.request.POST if is_custom_data_post else None,
```
Comment (czue) [inline corehq/apps/users/views/mobile/users.py]: this is the bugfix - when you posted from the commtrack tab it didn't have the CUD context and so was overriding the UI with a dict that had no values in it.

**Comment by benrudolph** [comment]
:white_check_mark: will wait for build

**Comment by czue** [comment]
failure didn't look related. kicked.

File: corehq/apps/users/views/mobile/users.py (hunk 2 above)
Comment (esoergel) [inline corehq/apps/users/views/mobile/users.py]: :+1: good catch

**Pull request merged**